Uabc 238 view attendees list

tulza opened the pull request:
closes #238

midnightcuberx committed:
Add basic wrappers

midnightcuberx committed:
Add basic wrappers

annabelleding committed:
make login page responsive

annabelleding committed:
minor changes

tulza committed:
Add view button

dyzhuu committed:
refactor wrappers

dyzhuu committed:
refactor wrappers

dyzhuu committed:
fix: types
* added types for wrappers
* installed @types/pg

dyzhuu committed:
fix: types
* added types for wrappers
* installed @types/pg

dyzhuu committed:
feat: wrap all existing api routes

dyzhuu committed:
lint

dyzhuu committed:
lint

dyzhuu committed:
fix: added correct return type to wrappers

dyzhuu committed:
fix: re-add rate-limit into verification code resending

annabelleding committed:
modify page layout for desktop view

annabelleding committed:
modify page layout for desktop view

dyzhuu committed:
fix: styling bugs & errors on mobile

dyzhuu committed:
feat: add exponential backoff to email sending

annabelleding committed:
fix issue of hidden components

annabelleding committed:
fix issue of hidden components

annabelleding committed:
minor edits

annabelleding committed:
minor edits

tulza committed:
Create game-sessions attendees endpoint

- route /api/game-sessions/[gameSessionId]/attendees
- basically a clone of download lol

tulza committed:
Add react query for booking destails data

tulza committed:
Update view session skeleton

tulza committed:
Attendees Table no pagination

dyzhuu committed:
added REPLY_TO address

LiamParker1 committed:
Remove email verified query parameter

LiamParker1 committed:
Add limit and offset as optional query parameters

dyzhuu committed:
feat: add playLevel field to user

LiamParker1 committed:
Check that the input integers are valid

dyzhuu committed:
feat: reword emails

dyzhuu committed:
feat: edit release workflow

dyzhuu committed:
fix: release workflow

dyzhuu committed:
fix: release workflow

dyzhuu committed:
chore: rename workflow

dyzhuu committed:
feat: improved api responses
* added standard api responses
* refactored relevant existing apis
* refactored route wrappers
* implemented better error toasts in booking & semesters page
* update lock file
* linted
* remove non-null assertions

dyzhuu committed:
feat: improved api responses
* added standard api responses
* refactored relevant existing apis
* refactored route wrappers
* implemented better error toasts in booking & semesters page
* update lock file
* linted
* remove non-null assertions

tulza committed:
Add date Display on view attendees list

- Add gameSession fetch
- Moved most of viewsession to client

tulza committed:
Add mobile responsiveness to attendees

tulza committed:
Add accordian to attendee list

tulza committed:
Download attendees list button

tulza committed:
Add Table sorting functionality + ts fix

dyzhuu commented on src/app/admin/view-sessions/[gameSessionId]/client-page.tsx:
```diff
@@ -0,0 +1,66 @@
+              onClick={handleDownloadcsv}
+            >
+              Download attendees list
+            </Button>
```
could you change this button to be the outline variant.

Change text to be "Download as CSV"

optionally, adding a download logo to the right of the text would be nice
https://lucide.dev/icons/download

dyzhuu commented on src/app/admin/@modal/(.)view-sessions-test/[gameSessionId]/page.tsx:
remove test file?

dyzhuu commented on src/components/admin/view-sessions/AdminViewSessionCard.tsx:
```diff
@@ -108,13 +109,29 @@ export function AdminViewSessionCard({
-        Download attendees list
-      </Button>
+      <div className="grid gap-2 sm:grid-cols-2">
+        <Button
+          className="w-full font-semibold"
+          disabled={attendees === 0}
+          onClick={handleButtonClick}
+        >
+          Download attendees list
+        </Button>
```
Remove this button from the card, and all associated functions. It should just be the View Attendees List button by itself

dyzhuu commented on src/components/admin/view-sessions/SkeletonViewSessionCard.tsx:
```diff
@@ -14,6 +21,9 @@ export const SkeletonViewSessionCard = () => (
       <Skeleton className="h-4 w-[150px] bg-tertiary/20" />
     </div>
-    <Skeleton className="h-10 w-full bg-tertiary/20" />
+    <div className="grid gap-2 sm:grid-cols-2">
+      <Skeleton className="h-10 w-full bg-tertiary/20" />
+      <Skeleton className="h-10 w-full bg-tertiary/20" />
+    </div>
```
revert per my other comment

dyzhuu commented on src/components/admin/view-sessions/gameSessionId/AttendeesList.tsx:
```diff
@@ -0,0 +1,180 @@
+
+const sortByMember = (a: AttendeesListResponse) => (a.member ? -1 : 1);
+
+const sortByPlayLevel = (a: AttendeesListResponse, b: AttendeesListResponse) =>
+  a.playLevel.localeCompare(b.playLevel);
```
This should sort by playLevel, not by alphabetical order. 

order by:
```
{
      advanced: 1,
      intermediate: 2,
      beginner: 3,
};
```

dyzhuu commented on src/components/admin/view-sessions/gameSessionId/AttendeesTablePagination.tsx:
delete empty file

dyzhuu commented on src/components/admin/view-sessions/AdminViewSessionCard.tsx:
```diff
@@ -108,13 +109,29 @@ export function AdminViewSessionCard({
+        >
+          <Button
+            className="w-full font-semibold"
+            variant="outline"
```
can we style the "View attendees list" button as primary variant

dyzhuu committed:
feat: modify NavigationBar functionality

tulza committed:
Resolve convesations

- Removed unnecesary files
- Removed download options from view sessions
- minor styles and text changes

dyzhuu committed:
fix: build & minor errors

dyzhuu committed:
fix: e2e pipeline

dyzhuu committed:
Merge branch 'main' into UABC-238-View-attendees-list
